src/voter/mod.rs

```diff
@@ -420,6 +423,84 @@ fn instantiate_last_round<H, N, E: Environment<H, N>>(
 	}
 }
 
+/// Allows querying the state of the voter.
+pub trait VoterState<Id: Eq + std::hash::Hash> {
+	fn voter_state(&self) -> report::VoterState<Id>;
+}
+
+pub mod report {
+	use std::collections::{HashMap, HashSet};
+	use crate::weights::{VoteWeight, VoterWeight};
+
+	/// Basic data struct for the state of a round.
+	#[derive(PartialEq, Eq, Clone)]
+	#[cfg_attr(test, derive(Debug))]
+	pub struct RoundState<Id: Eq + std::hash::Hash> {
+		pub total_weight: VoterWeight,
+		pub threshold_weight: VoterWeight,
```

Comment:
missing docs on these fields, `VoterState` fields, and the `mod report`, and `fn voter_state`. There are even more missing docs - remember the guideline that all public items should be documented.